packages/cojson/src/storage/sqlite/client.ts

```diff
@@ -193,4 +194,40 @@ export class SQLiteClient
     this.db.transaction(() => operationsCallback(this));
     return undefined;
   }
+
+  getUnsyncedCoValueIDs(): RawCoID[] {
+    const rows = this.db.query<{ co_value_id: RawCoID }>(
+      "SELECT DISTINCT co_value_id FROM unsynced_covalues",
+      [],
+    ) as { co_value_id: RawCoID }[];
+    return rows.map((row) => row.co_value_id);
+  }
+
+  trackCoValuesSyncState(
+    operations: Array<{ id: RawCoID; peerId: PeerID; synced: boolean }>,
+  ): void {
+    if (operations.length === 0) {
+      return;
+    }
+
+    this.db.transaction(() => {
```

Comment:
Do we need to run these inside a transaction?

Reply:
Good point, we don't